etl/data_management/DataManager.py

```diff
@@ -0,0 +1,214 @@
+import os
+import time
+import datetime
+import shutil
+from shared.LoggerFactory import LoggerFactory
+from shared.utils import read_data_from_json, write_data_to_json
+
+ROOT_DIR = os.environ['ROOT_DIR']
+
+
+class DataManager:
+    """ Class that collects the enhanced data from the scraping process, manages backups and composes the data that is
+     will be uploaded to elasticsearch
+     - After the enhancement process, the results are stored in a backup folder named after the time of the backup
+     - the upload folder contains the data that will be uploaded to elasticsearch
+
+     For the upload data, the most recent backup data is selected. The amount of posts in the selected dataset is
+     compared against older backups according to the value set in fallback_depth. A fallback_depth of 2 means that the
+     data from the last 2 backups that are older than the current selected backup is compared with the upload data.
+     If the current dataset for the upload contains less than X% of the posts in the backup dataset, where X is the
+     defined threshold, the backup dataset is selected instead. This is done on a file by file basis, meaning the
+     upload can contain a mixture of files from different backups.
+
+     The source of all files in the upload folder at the end of the process gets logged."""
+
+    # manages how many backups are to be kept. if the number of existing backups would exceed this threshold, the oldest
+    # backup gets deleted
+    max_number_of_backups = 7
+
+    # manages how many backups into the past should be considered for the upload
+    fallback_depth = 3
+
+    # defines the percentage threshold at which data from an older backup may be used. data from an older backup may be
+    # used if the current data selected for upload contains less than [threshold] * [number of posts in backup], the
+    # data from the backup is selected for upload instead
+    threshold = 0.75
+
+    enhanced_data_location = os.path.join(ROOT_DIR, ROOT_DIR, 'data_enhancement/data')
```

Comment:
this was indeed an error that was not immediately noticeable because ROOT_DIR is not a joinable path but a complete one and thus overrides the previous joined ROOT_DIR. it is fixed now

Reply:
this was indeed an error that was not immediately noticeable because ROOT_DIR is not a joinable path but a complete one and thus overrides the previous joined ROOT_DIR. it is fixed now